### PR TITLE
SSH support + QoL improvements (hopefully!)

### PR DESCRIPTION
**Pull request by NuSkooler**
🚧 WORK IN PROGRESS SSH SUPPORT 🚧
I can connect + handshake and display incoming data. I didn't have time to really look yet, but I think we're now in the recv blocking vs send (ie: need two threads) zone.

Try to detect dark vs light OS theme/mode & apply; default to dark. Let me know if this works for you. For me, I get the white lettering on dark in dark mode & black lettering on white in light mode.

Put a little note in there for discussion RE: TerminalType - if we allow for e.g. "ansi-bbs-256color", "xterm-truecolor", etc., some boards will try to use extended abilities.

### CODE REVIEW

File: src/com/ssh.rs [hunk 1]
```diff
@@ -1,79 +1,134 @@
...
+        sess.set_option(SshOption::Port(port))?;
+        // sess.set_option(SshOption::PublicKeyAcceptedTypes(
+        //     ACCEPTED_PK_TYPES.to_string(),
+        // ))?;
         sess.options_parse_config(None)?;
         sess.connect()?;
-        let auth_methods = sess.userauth_list(Some(&addr.user_name))?;
 
-        sess.userauth_agent(Some(&addr.user_name))?;
+        //  :TODO: SECURITY: verify_known_hosts() implemented here -- ie: user must accept & we save somewhere
```
Comment (NuSkooler) [inline src/com/ssh.rs:73]: We'll want to do something here & persist saved allowed host keys somewhere

File: src/com/ssh.rs [hunk 2]
```diff
@@ -1,79 +1,134 @@
...
+
+        sess.userauth_password(Some(addr.user_name.as_str()), Some(addr.password.as_str()))?;
+
         let chan = sess.new_channel()?;
-        let mut result = [0u8; 1024];
-        let size: usize = chan.read_timeout(&mut result, false, None)?;
+        chan.open_session()?;
+        let terminal_type = addr.terminal_type.to_string().to_lowercase();
+        chan.request_pty(terminal_type.as_str(), self.window_size.width as u32, self.window_size.height as u32)?;
```
Comment (NuSkooler) [inline src/com/ssh.rs:80]: does icy emit terminal size updates in any way? If/or if it will, we can plug a handler to send updates

**Comment by NuSkooler** [comment]
Looks like main changed again - when I merged in, I can no longer get the main window to show.

**Comment by mkrueger** [comment]
I simplified the threading a bit to get it work with wasm. 

The thread for open connection is gone. There is now always a 2nd thread for communication and OpenConnection is now done with a message to the thread. It's a work an progress as well. I'll take a look.

**Pull request merged**